src/ipc/ipc4/handler.c

```diff
@@ -1527,4 +1530,12 @@ void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 
 		ipc4_send_reply(&reply);
 	}
+
+	/*
+	 * When the first FW_GEN IPC has been processed we are in a stable
+	 * running state, now if a test causes an exception, we have a good
+	 * chance of capturing it.
+	 */
+	if (target == SOF_IPC4_MESSAGE_TARGET_FW_GEN_MSG)
+		run_once(ztest_run_test_suite, sof_boot);
```

Comment:
@lyakh  How does this work if we have boot test disabled in the build? Do we still run this?

Reply:
@kv2019i good question :-) You're right, have to make it conditional on `CONFIG_...`, thanks for spotting!